From 94096702376ecb99c86cbee9dd95fc3675231b8a Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Wed, 26 May 2021 15:25:40 +0900
Subject: [PATCH] cifsd: move fips_enabled check before the str_to_key()
Git-commit: 94096702376ecb99c86cbee9dd95fc3675231b8a
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Move fips_enabled check before the str_to_key().

Reviewed-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/auth.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/fs/cifsd/auth.c b/fs/cifsd/auth.c
index 6b90aac86fcc..9263c9ce2dd2 100644
--- a/fs/cifsd/auth.c
+++ b/fs/cifsd/auth.c
@@ -92,14 +92,13 @@ smbhash(unsigned char *out, const unsigned char *in, unsigned char *key)
 	unsigned char key2[8];
 	struct des_ctx ctx;
 
-	str_to_key(key, key2);
-
 	if (fips_enabled) {
 		ksmbd_debug(AUTH,
 			"FIPS compliance enabled: DES not permitted\n");
 		return -ENOENT;
 	}
 
+	str_to_key(key, key2);
 	des_expand_key(&ctx, key2, DES_KEY_SIZE);
 	des_encrypt(&ctx, out, in);
 	memzero_explicit(&ctx, sizeof(ctx));
-- 
2.33.0

